src/formatDate.js

```diff
@@ -8,7 +8,38 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[fromFormat.length - 1];
+  const toSeparator = toFormat[toFormat.length - 1];
+  const fromParts = fromFormat.slice(0, -1);
+  const toParts = toFormat.slice(0, -1);
+
+  const dateParts = date.split(fromSeparator);
+
+  const dateObj = {};
+
+  fromParts.forEach((part, index) => {
```

Comment:
It's good practice to validate the input data before processing it. Consider adding checks to ensure that 'date' is a valid string and that 'fromFormat' and 'toFormat' are valid arrays with the correct length. This can help prevent runtime errors and make the function more robust.